server/backend-tests/db.test.js

```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+
+beforeAll(async () => {
+  await mongoose.connect('mongodb://localhost/gameinfo', { useNewUrlParser: true, useUnifiedTopology: true });
+});
+
+describe('Mongoose', () => {
+  it('mongoose should connect to the database.', () => {
+    expect(mongoose.connection.readyState).toEqual(1);
+  });
+});
+
```

Comment:
This afterAll needs to be before the actual tests. All beforeEach, beforeAll, afterEach, and afterAll must be declared before the test you want them to work on. Otherwise that's never actually running on your tests. Everything there runs in order. It MAY run if your AJAX takes long enough, but it's anyone's guess. Just move your whole afterAll() block up a few lines and you're good.

Reply:
Good catch, thanks!